spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/tracing/prometheus/PrometheusExemplarsAutoConfigurationTests.java

```diff
@@ -105,24 +105,51 @@ void prometheusOpenMetricsOutputShouldContainExemplars() {
 				.map((matchResult) -> new TraceInfo(matchResult.group(2), matchResult.group(1)))
 				.findFirst();
 
-			Optional<TraceInfo> counterTraceInfo = openMetricsOutput.lines()
-				.filter((line) -> line.contains("test_observation_seconds_count") && line.contains("span_id"))
-				.map(COUNTER_TRACE_INFO_PATTERN::matcher)
-				.flatMap(Matcher::results)
-				.map((matchResult) -> new TraceInfo(matchResult.group(2), matchResult.group(1)))
-				.findFirst();
-
-			assertThat(bucketTraceInfo).isNotEmpty().contains(counterTraceInfo.orElse(null));
+			assertThat(bucketTraceInfo).isNotEmpty();
 		});
 	}
 
+	@Test
+	void prometheusOpenMetricsOutputCanBeConfiguredToContainExemplarsOnHistogramCount() {
+		this.contextRunner.withSystemProperties("io.prometheus.exporter.exemplarsOnAllMetricTypes=true")
```

Comment:
The Prometheus client does not add exemplars to the count of a histogram by default. This configuration is needed to restore that behavior that Micrometer had previously. We are in discussion about a way to expose this configuration via Micrometer. You can see more about it in the Prometheus client documentation: https://prometheus.github.io/client_java/config/config/

Reply:
Since https://github.com/micrometer-metrics/micrometer/pull/4921 and the corresponding updates I made to this PR in https://github.com/spring-projects/spring-boot/pull/40023/commits/63a3d2a4ee413f8e24faa25735d284029057f19c things are back to having exemplars on the histogram count by default (when using Micrometer). Configuration can be done via Micrometer's PrometheusConfig and consequently Boot's PrometheusProperties now.